### DISCUSSION TITLE
C# pre and postprocessing with opencvsharp for onnxruntime

### DISCUSSION THREAD

**Discussion by andife** [microsoft/onnxruntime#11411] [Other Q&A]
Hello,

I would be quite interested if someone could share code with preprocessing the data with opencvsharp and maybe even more
postprocessing the output data. My use case would be semantic segmentation with different output layer. So far I've not found code snippets using #C and opencv. For Preprocessing the main question is how to create the input tensor in NCHW format and than at the output get access to the different output layer.

**Comment by lionkinging** [comment]
I also encountered the same problem. Can we communicate together

**Comment by GeorgeS2019** [comment]
related issues:
- [ ] [[C#] ContribOperators](https://github.com/microsoft/onnxruntime/issues/11564)
- [ ] [[Documentation, Windows, c#] Add an entry on Build for Custom Op development](https://github.com/microsoft/onnxruntime/issues/11657)
- [ ] [[C#] Documentation on how to coordinate the external Onnxruntime-Extension](https://github.com/microsoft/onnxruntime/issues/11655)
- [ ] [[C#] Basic build instruction to generate custom_op_library.dll](https://github.com/microsoft/onnxruntime/issues/11655)
- [ ] [[C#] Provide custom_op_library.dll](https://github.com/microsoft/onnxruntime/issues/11653)

**Comment by GeorgeS2019** [comment]
@andife and @lionkinging 

opencvsharp are pre and postprocessing contributed operators and it seems currently there is no easy way to access them in c# 

See the discussion [here](https://github.com/microsoft/onnxruntime/discussions/11726)
